src/main/java/ca/on/oicr/gsi/dimsum/service/filtering/QCStatusSort.java

```diff
@@ -0,0 +1,46 @@
+package ca.on.oicr.gsi.dimsum.service.filtering;
+
+import java.util.Comparator;
+import java.util.List;
+import ca.on.oicr.gsi.cardea.data.Sample;
+
+public class QCStatusSort implements Comparator<Sample> {
+
+    private boolean reverseOrder;
+
+    public void setReverseOrder(boolean reverseOrder) {
+        this.reverseOrder = reverseOrder;
+    }
+
+    public static final List<String> QC_STATUS_ORDER =
+            List.of("pendingQC", "dataReview", "topUp", "passed", "failed");
+
+    public static String getSampleQcStatus(Sample sample) {
+        if (PendingState.Helpers.isPendingQc(sample)) {
+            return "pendingQC";
+        } else if (PendingState.Helpers.isPendingDataReview(sample)) {
+            return "dataReview";
+        } else if (PendingState.Helpers.isTopUpRequired(sample)) {
+            return "topUp";
+        } else if (PendingState.Helpers.isPassed(sample)) {
+            return "passed";
+        } else {
+            return "failed";
+        }
+    }
```

Comment:
Can simplify this by returning integers instead. Then there's no need for the Strings, the order collection, or lookup in your `compare` method - just compare the values returned from this method using `Integer.compare`